File: abc/LockerDoors/src/Lambda.go
package main

import (
    "fmt"
    "math"
)

func bruteForce(n int) []bool {
    arr := make([]bool, n)

    for i := 1; i <= n; i++ {
        for j := i - 1; j < n; j += i {
            arr[j] = !arr[j]
        }
    }

    return arr
}

func process(n int) []bool {
    arr := make([]bool, n)

    var temp int
    for i := 1; i <= n; i++ {
        temp = int(math.Sqrt(float64(i)))
        arr[i - 1] = temp * temp == i
    }

    return arr
}

func main() {
    const N = 100

    arr := process(N)

    for i, v := range arr {
        if v {
            fmt.Printf("%d\t  OPEN\n", i + 1)
        }
    }
}

File: abc/LockerDoors/src/Lambda.py
import math

def bruteForce(n: int) -> list:
    arr = [False] * n

    for i in range(1, n + 1):
        for j in range(i - 1, n, i):
            arr[j] = not arr[j]

    return arr

def process(n: int) -> list:
    arr = [False] * n

    temp = 0
    for i in range(1, n + 1):
        temp = int(math.sqrt(float(i)))
        arr[i - 1] = temp * temp == i

    return arr

def main() -> None:
    N = 100

    arr = process(N)

    for i, v in enumerate(arr):
        if v:
            print("%d\t  OPEN" % (i + 1))

if __name__ == "__main__":
    main()
